### PR TITLE
DScanner parser added

### PR DESCRIPTION
**Pull request by andre2007**
Parser for DScanner (https://github.com/dlang-community/D-Scanner) added

### CODE REVIEW

**Comment by andre2007** [comment]
@uhafner why the checks are failing, what is missing?

**Comment by uhafner** [comment]
Ah, you need to increase the version to 5.2.0-SNAPSHOT since you are adding a new feature.

**Comment by andre2007** [comment]
@uhafner every line and and branch is covered by tests. I do not understand what codecov is complaining about.

**Comment by uhafner** [comment]
Well, that tool is somewhat niggling since I use this project as reference in my testing lecture ;-) I think it noticed that the branch coverage could be improved since all `if` statements in `convertToIssue` have only a test with `true` but not with `false`. You can improve that coverage by writing an additional test, that has a file with only the filename property for an issue in the JSON file.

File: src/main/java/edu/hm/hafner/analysis/parser/DScannerParser.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import edu.hm.hafner.analysis.ReaderFactory;
+import edu.hm.hafner.analysis.Report;
+import edu.hm.hafner.analysis.Severity;
+
+/**
+ * Parser report in JSON format as exported by DScanner.
+ *
+ * @author Andre Pany
+ */
+public class DScannerParser extends JsonBaseParser {
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/DScannerParser.java]: Would it rather make sense to derive directly from `JsonParser`? Then we can reuse the methods `accepts` and `parse`?

File: src/main/java/edu/hm/hafner/analysis/parser/DScannerParser.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            builder.setColumnStart(jsonIssue.getInt(COLUMN));
+        }
+        if (jsonIssue.has(MESSAGE)) {
+            builder.setMessage(jsonIssue.getString(MESSAGE));
+        }
+        return builder.buildOptional();
+    }
+
+    private Severity getSeverityByKey(final String key) {
+        String[] parts = key.split("\\.");
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/DScannerParser.java]: Would make more safe to use ` key.split("\\.", -1);`, see https://ci.jenkins.io/job/Plugins/job/analysis-model/view/change-requests/job/PR-189/5/error-prone/new/source.bd37f77a-752a-4696-a046-73324be6f7db/#83.

File: src/test/java/edu/hm/hafner/analysis/parser/DScannerParserTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import edu.hm.hafner.analysis.ParsingException;
+import edu.hm.hafner.analysis.Report;
+import edu.hm.hafner.analysis.Severity;
+import edu.hm.hafner.analysis.assertj.SoftAssertions;
+
+import static edu.hm.hafner.analysis.assertj.Assertions.*;
+
+/**
+ * Tests the class {@link DScannerParser}.
+ */
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/parser/DScannerParserTest.java]: If you like you can add \@author here as well

**Comment by andre2007** [comment]
@uhafner all done, please check

**Comment by uhafner** [comment]
Thank you very much!

**Pull request merged**